app/gui2/src/components/GraphEditor.vue

```diff
@@ -21,12 +22,24 @@ const viewportNode = ref<HTMLElement>()
 const navigator = useNavigator(viewportNode)
 const graphStore = useGraphStore()
 const projectStore = useProjectStore()
+const executionCtx = ref<Opt<ExecutionContext>>()
```

Comment:
Could this be a `shallowRef` instead? Also, `Opt` is not needed when declaring `ref`s, as the `T | undefined` type is implicit when not providing the initial value.

```suggestion
const executionCtx = shallowRef<ExecutionContext>()
```